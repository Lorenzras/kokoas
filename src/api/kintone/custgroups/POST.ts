import { APP_ID } from './config';
import { KintoneRecord } from './../config';

import { AddRecordFn, UpdateRecordParam } from '../restapi';


export const addCustGroup: AddRecordFn = (record = {}) => {

  return KintoneRecord.addRecord({ app: APP_ID, record });

};

export const updateGroup = async (group: UpdateRecordParam) => {
  return KintoneRecord.updateRecord({
    app: APP_ID,
    id: group.id,
    record: group.record,
    revision: group.revision,
  });
};






