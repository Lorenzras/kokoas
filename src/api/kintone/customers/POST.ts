import { APP_ID } from './config';
import { KintoneRecord } from './../config';

import { AddRecordResult, UpdateRecordParam, RecordParam } from '../restapi';



export const addCustomer = (record: RecordParam = {}): Promise<AddRecordResult> => {

  return KintoneRecord.addRecord({ app: APP_ID, record });

};

/**
 * Adds multiple customers to an app.
 *
 * @param records
 * @returns an array of objects that include id and revision of created records.
 */
export const addCustomers = async (records: RecordParam[]) => {
  return KintoneRecord.addRecords({ app: APP_ID, records });
};



export const updateCustomers = async (records : UpdateRecordParam[]) => {
  return KintoneRecord.updateRecords({
    app: APP_ID,
    records,
  });
};


