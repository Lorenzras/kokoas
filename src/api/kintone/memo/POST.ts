import { KintoneRecord } from '../config';
import { RecordParam } from '../restapi';
import { APP_ID } from './config';




export const addMemo = (record: CustomerMemoTypes.SavedData  ) => {

  return KintoneRecord.addRecord({ app: APP_ID, record: record as unknown as RecordParam });
};