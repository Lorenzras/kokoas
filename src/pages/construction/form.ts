import * as Yup from 'yup';

export type BuildingTypeVals =
| '戸建て'
| 'マンション'
| '店舗/事務所'
| 'その他';

export type KeyOfConstructionDetails = keyof typeof initialValues;
export type ConstructionDetailsValues = Partial<Record<KeyOfConstructionDetails, string | number | boolean>>;

/**
 * Set Initial values here in case MUI is shouting about un/controlled components.
 */
export const initialValues = {
  id: '',
  constructionTypeId: '',
  constructionName: '',
  isAgentConfirmed: false,
  cocoConst1: '',
  cocoConst2: '',
  postal: '',
  address1: '',
  address2: '',
  addressKari: '',
  custGroupId: '',
  buildingType: '戸建て' as BuildingTypeVals,
  isChkAddressKari: false,
};

/**
 * Set Validation for fields that requires it.
 * Refer to YUM documentation.
 */
export const validationSchema =  Yup.object(
  {
    custGroupId: Yup
      .string()
      .required('必須です。'),
    postal: Yup
      .string()
      .required('必須です。'),
  } as Partial<Record<KeyOfConstructionDetails, any>>,
);
