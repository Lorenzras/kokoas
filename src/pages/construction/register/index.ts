export * from './ConstructionForm';
export * from './form';
