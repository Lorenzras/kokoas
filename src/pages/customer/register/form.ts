import * as Yup from 'yup';



export type KeyOfConstructionDetails = keyof ConstructionDetails.SavedData;
export type ConstructionDetailsValues = Partial<Record<KeyOfConstructionDetails, string | number | boolean>>;

enum CustomerFieldNames  {
  'custName',
  'custNameReading',
  'gender',
  'birthYear',
  'birthMonth',
  'birthDay',
  'postal',
  'address1',
  'address2',
  'phone1',
  'phone1Type',
  'phone2',
  'phone2Type',
  'email',
  'isSameAddress',
}

type Customer = {
  [K in keyof typeof CustomerFieldNames ]: string | boolean;
};

export const initialCustomerValue: Customer = {
  custName: '',
  isSameAddress: true,
  custNameReading: '',
  gender: '',
  birthYear: '',
  birthMonth: '',
  birthDay: '',
  postal: '',
  address1: '',
  address2: '',
  phone1: '',
  phone1Type: '',
  phone2: '',
  phone2Type: '',
  email: '',
};

/**
 * Set Initial values here in case MUI is shouting about un/controlled components.
 */
export const initialValues = {
  store: '',
  cocoAG1: '',
  cocoAG2: '',
  yumeAG1: '',
  yumeAG2: '',
  customers: [initialCustomerValue],
};

export type CustomerForm = typeof initialValues;
export type CustomerFormKeys = keyof CustomerForm;
export type  CustomerInstanceKeys = (keyof typeof CustomerFieldNames);

/**
 * Set Validation for fields that requires it.
 * Refer to YUM documentation.
 */
export const validationSchema =  Yup.object(
  {
    custGroupId: Yup
      .string()
      .required('必須です。'),
    postal: Yup
      .string()
      .required('必須です。'),
  } as Partial<Record<KeyOfConstructionDetails, any>>,
);


export const getFieldName = (fieldName : CustomerInstanceKeys) => fieldName;