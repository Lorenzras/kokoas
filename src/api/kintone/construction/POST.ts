import { ConstructionDetailsValues } from '../../../pages/construction/register/form';
import { APPIDS, KintoneRecord } from '../config';

export const convertToKintone = (rawValues: ConstructionDetailsValues) => {
  return Object.entries(rawValues)
    .reduce((acc, [key, rawVal])=>{
      return { ...acc, [key]: { value: rawVal } };
    }, {});

};

export const saveConstructionData = (rawValues: ConstructionDetailsValues) =>{
  const record = convertToKintone(rawValues);
  return KintoneRecord.addRecord({ app: APPIDS.constructionDetails, record });
};
