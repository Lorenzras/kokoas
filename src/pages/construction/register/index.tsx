
import MainContainer from '../../../components/ui/containers/MainContainer';
import PageTitle from '../../../components/ui/labels/PageTitle';
import ConstructionInfo from './sections/ConstructionInfo';
import CustInfo from './sections/CustInfo';
import { Foot, Submit } from './sections/bottom';
import ConstructionLocation from './sections/location';
import { Divider, Grid } from '@mui/material';

import { Formik, Form } from 'formik';

import { validationSchema, initialValues } from './form';
import { saveConstructionData } from '../../../api/kintone/construction/POST';


const ConstructionRegister  = () => {

  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={(values, { setSubmitting }) => {
        saveConstructionData(values)
          .then((resp)=>{
            console.log('SAVED!', resp);
            setSubmitting(false);
          });
      }}
    >
      <Form>
        <MainContainer>
          <PageTitle label="工事情報登録" color="#60498C" textColor='#FFF' />
          <CustInfo/>
          <ConstructionInfo />
          <ConstructionLocation/>

          <Grid item xs={12}><Divider/></Grid>

          <Submit/>
          <Foot/>
        </MainContainer>
      </Form>
    </Formik>

  );
};

export default ConstructionRegister;