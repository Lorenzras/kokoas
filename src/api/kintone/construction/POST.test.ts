import 'regenerator-runtime/runtime';
import { convertToKintone, saveConstructionData } from './POST';
import { initialValues } from './../../../pages/construction/register/form'; //'src/pages/construction/register/form';

describe('Raw Values', ()=>{
  it('is converterd', ()=>{
    expect(convertToKintone(initialValues)).toMatchSnapshot();
  });

  it('is saved', ()=>{
    console.log(process.env.API_CONSTRUCTION_TYPE);
    expect(saveConstructionData(initialValues)).toMatchSnapshot();
  });
});
