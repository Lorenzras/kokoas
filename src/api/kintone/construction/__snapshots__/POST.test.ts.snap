// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Raw Values is converterd 1`] = `
Object {
  "address1": Object {
    "value": "",
  },
  "address2": Object {
    "value": "",
  },
  "addressKari": Object {
    "value": "",
  },
  "agent1Id": Object {
    "value": "",
  },
  "agent2Id": Object {
    "value": "",
  },
  "buildingType": Object {
    "value": "戸建て",
  },
  "constructionName": Object {
    "value": "",
  },
  "constructionTypeId": Object {
    "value": "",
  },
  "custGroupId": Object {
    "value": "",
  },
  "isAgentConfirmed": Object {
    "value": false,
  },
  "isChkAddressKari": Object {
    "value": false,
  },
  "postal": Object {
    "value": "",
  },
}
`;

exports[`Raw Values is saved 1`] = `Promise {}`;
